e2e/rdctl.e2e.spec.ts

```diff
@@ -787,13 +790,74 @@ test.describe('Command server', () => {
           expect(stderr).toEqual('');
           expect(JSON.parse(stdout)).toEqual([
             'GET /v0',
+            'GET /v0/diagnostic_categories',
+            'GET /v0/diagnostic_check',
+            'GET /v0/diagnostic_ids',
             'PUT /v0/factory_reset',
             'PUT /v0/propose_settings',
             'GET /v0/settings',
             'PUT /v0/settings',
             'PUT /v0/shutdown',
           ]);
         });
+        test('/v1 should fail', async() => {
+          const { stdout, stderr } = await rdctl(['api', '/v1']);
+
+          expect({ stdout: JSON.parse(stdout), stderr: stderr.trim() }).toMatchObject({ stdout: { message: '404 Not Found' }, stderr: 'Unknown command: GET /v1' });
+        });
+
+        test.describe('diagnostics', () => {
+          let categories: string[];
+
+          test('categories', async() => {
+            const { stdout, stderr } = await rdctl(['api', '/v0/diagnostic_categories']);
+
+            expect(stderr).toEqual('');
+            categories = JSON.parse(stdout);
+            expect(categories).toMatchObject(['Utilities', 'Networking']);
+          });
+          test('it finds the IDs for Utilities', async() => {
+            const { stdout, stderr } = await rdctl(['api', '/v0/diagnostic_ids?category=Utilities']);
```

Comment:
Looking at the actual API call, I suspect we should have category IDs instead of using the label directly. We can deal with that later, though, I think — we're not guaranteeing API compatibility yet.

Reply:
I figured the labels can act as IDs for now, until we localize them.